klib/syslog.c

```diff
@@ -327,11 +343,18 @@ define_closure_function(0, 2, void, syslog_timer_func,
                         u64, expiry, u64, overruns)
 {
     if (overruns != timer_disabled) {
-        syslog_file_flush();
+        syslog_file_flush(0);
         syslog_udp_flush();
     }
 }
 
+define_closure_function(0, 2, void, syslog_shutdown_completion,
+                        int, status, merge, m)
+{
+    syslog_file_flush(apply_merge(m));
+    syslog_udp_flush();
```

Comment:
I think it's theoretically possible for the final completion from the merge (which shuts off the vm) to run before the udp flush, so maybe the udp flush should be first here?

Reply:
The completion invoked by syslog_file_flush() can theoretically run before the udp flush, but the final completion from the merge won't run until after all shutdown completions are applied: kernel_shutdown() keeps the merge open until the call to apply(sh, STATUS_OK).

Reply:
Right, I see. Looks good!